### PR TITLE
Test boundary extrude

### PR DESCRIPTION
**Pull request by lukasdreyer**
**_Describe your changes here:_**
Add a new test for consistency of creating the boundary element and extruding it again.
Depends on https://github.com/DLR-AMR/t8code/pull/854

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: test/t8_schemes/t8_gtest_boundary_extrude.cxx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  /* For elements that are on the face of the root element, check that creating the boundary element
+   * and extruding it results in the original element
+    */
+  virtual void
+  check_element ()
+  {
+    const int num_faces = ts->t8_element_num_faces (element);
+    for (int iface = 0; iface < num_faces; iface++) {
+      /* Iterate over all faces that are also root faces and determine the face element */
+      if (!ts->t8_element_is_root_boundary (element, iface))
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_boundary_extrude.cxx:40]:
```suggestion
      if (ts->t8_element_is_root_boundary (element, iface)){
```

File: test/t8_schemes/t8_gtest_boundary_extrude.cxx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        continue;
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_boundary_extrude.cxx:41]:
```suggestion
```

File: test/t8_schemes/t8_gtest_boundary_extrude.cxx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      t8_element_t *boundary;
+      face_ts->t8_element_new (1, &boundary);
+
+      ts->t8_element_boundary_face (element, iface, boundary, face_ts);
+
+      ts->t8_element_extrude_face (boundary, face_ts, check, tree_face);
+
+      EXPECT_ELEM_EQ (ts, element, check);
+
+      face_ts->t8_element_destroy (1, &boundary);
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_boundary_extrude.cxx:57]:
```suggestion
      face_ts->t8_element_destroy (1, &boundary);
      }
```

**Review by Davknapp — CHANGES_REQUESTED** [review]
Please have a look at my suggestions. Also update the indentation if you use them

**Review by Davknapp — APPROVED** [review]
